src/main/java/com/sofrelgroup14/ci/controller/BuildController.java

```diff
@@ -1,34 +1,130 @@
 package com.sofrelgroup14.ci.controller;
+import java.io.BufferedReader;
+import java.io.File;
 
+// For parsing JSON-strings (source: https://stackoverflow.com/a/5245881)
+import org.json.JSONException;
+import org.json.JSONObject;
 
+import org.apache.tomcat.jni.Proc;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import com.sofrelgroup14.ci.models.Logs;
 import com.sofrelgroup14.ci.repositories.LogsRepository;
 import org.bson.types.ObjectId;
+
+import java.io.InputStreamReader;
 import java.util.List;
 import java.time.Instant;
 
+/**
+ * Rest controller for the /build-endpoint, invoked by the
+ * GitHub webhook when a push is made to a branch.
+ */
 @RestController
 @RequestMapping("/build")
 public class BuildController {
     
     @Autowired
     private LogsRepository repository;
 
-
+    /*
+    Sources used to learn how to extract the HTTP-body from the POST-request
+    made by the GitHub webhook. (I.e. how to use @RequestBody)
+    1) https://stackabuse.com/get-http-post-body-in-spring/
+    2) https://www.baeldung.com/spring-request-response-body
+    3) https://stackoverflow.com/a/6019761
+    */
+    /**
+     * Builds the branch specified in "ref" attribute of the webhook payload
+     * (this is the branch that was pushed to), runs tests on code in this branch,
+     * and finally logs the output to the database.
+     *
+     * Does this by cloning the repo, checking out the branch, building
+     * and running tests using 'mvn' command, and then deletes the repo.
+     *
+     * Example payload (irrelevant parts omitted):
+     * {
+     *  ...
+     *  "ref": "refs/heads/webhooks-playground",
+     *  "before": "c130af7052edd8cc063dd28df6441262cb782822",
+     *  "after": "bee5e5bd2e8da4bcde79f8a0483bda650bdb1b95",
+     *  ...
+     * }
+     * would run:
+     * 'git clone -b webhooks-playground <repo-url> repo'
+     * 'mvn verify --file repo/pom.xml'
+     * ...save the output to the database...
```

Comment:
Regarding the \<repo-url\>, I guess it's OK to hard code it? In this case we have `https://github.com/sofrel-group14/ci.git` (further down, line 95) but I guess in a "real" application that could be an env variable or something. As far as I'm concerned we can keep it like that for this assignment, I don't think they'll mind.

Reply:
I think Cyrille said in the lecture that it could be hard coded

Reply:
I agree, we can keep it like this